Sources/MapboxMaps/Location/Puck/Puck3D.swift

```diff
@@ -89,15 +95,69 @@ internal final class Puck3D: Puck {
             try! style.addSource(source, id: Self.sourceID)
         }
 
+        // Mercator scale
+        maybeUpdateMercatorScale(at: location.coordinate.latitude)
+
         // create the layer if needed
         if !style.layerExists(withId: Self.layerID) {
             var modelLayer = ModelLayer(id: Self.layerID)
             modelLayer.source = Self.sourceID
             assert(modelLayer.paint != nil)
             modelLayer.paint?.modelLayerType = .constant(.locationIndicator)
-            modelLayer.paint?.modelScale = configuration.modelScale
             modelLayer.paint?.modelRotation = configuration.modelRotation
+            updateModelScale(layer: &modelLayer)
             try! style.addPersistentLayer(modelLayer, layerPosition: nil)
         }
     }
+
+    private func maybeUpdateMercatorScale(at latitude: Double) {
```

Comment:
The `IfNeeded` naming convention would be a good fit here: `updateModelScaleIfNeeded`